fuel/datasets/hdf5.py

```diff
@@ -119,7 +119,8 @@ class H5PYDataset(Dataset):
         indices are ordered.
 
     """
-    ref_counts = defaultdict(int)
+    _ref_counts = defaultdict(int)
+    _file_handles = defaultdict()
```

Comment:
No need to make this a `defaultdict`. The reason why `_ref_counts` is a `defaultdict` is to simplify the logic in the defunct `_get_file_id` method, but for `_file_handles` we gain nothing.